provider/lxd/server_integration_test.go

```diff
@@ -0,0 +1,550 @@
+		Addresses: []string{
+			"https://192.168.0.1:8443",
+		},
+	}
+	serverInfo := &api.Server{
+		ServerUntrusted: api.ServerUntrusted{
+			APIVersion: "1.1",
+		},
+	}
+
+	factory, server, interfaceAddr := lxd.NewLocalServerFactory(ctrl)
+
+	gomock.InOrder(
+		server.EXPECT().GetProfile("default").Return(profile, etag, nil),
+		server.EXPECT().VerifyNetworkDevice(profile, etag).Return(nil),
+		server.EXPECT().EnableHTTPSListener().Return(nil),
+		server.EXPECT().LocalBridgeName().Return(bridgeName),
+		interfaceAddr.EXPECT().InterfaceAddress(bridgeName).Return(hostAddress, nil),
+		server.EXPECT().GetConnectionInfo().Return(connectionInfo, nil),
+		server.EXPECT().StorageSupported().Return(true),
+		server.EXPECT().GetProfile("default").Return(profile, etag, nil),
+		server.EXPECT().EnsureDefaultStorage(profile, etag).Return(nil),
+		server.EXPECT().GetServer().Return(serverInfo, etag, nil),
+	)
+
+	svr, err := factory.RemoteServer(environscloudspec.CloudSpec{
+		Endpoint: "",
+	})
+	c.Assert(svr, gc.Not(gc.IsNil))
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *serverIntegrationSuite) TestRemoteServer(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	profile := &api.Profile{}
+	etag := "etag"
+	serverInfo := &api.Server{
+		ServerUntrusted: api.ServerUntrusted{
+			APIVersion: "1.1",
+		},
+	}
+
+	factory, server := lxd.NewRemoteServerFactory(ctrl)
+
+	gomock.InOrder(
+		server.EXPECT().StorageSupported().Return(true),
+		server.EXPECT().GetProfile("default").Return(profile, etag, nil),
+		server.EXPECT().EnsureDefaultStorage(profile, etag).Return(nil),
+		server.EXPECT().GetServer().Return(serverInfo, etag, nil),
+	)
+
+	creds := cloud.NewCredential("any", map[string]string{
+		"client-cert": "client-cert",
+		"client-key":  "client-key",
+		"server-cert": "server-cert",
+	})
+	svr, err := factory.RemoteServer(environscloudspec.CloudSpec{
+		Endpoint:   "https://10.0.0.9:8443",
+		Credential: &creds,
+	})
+	c.Assert(svr, gc.Not(gc.IsNil))
+	c.Assert(svr, gc.Equals, server)
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *serverIntegrationSuite) TestRemoteServerWithNoStorage(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	etag := "etag"
+	serverInfo := &api.Server{
+		ServerUntrusted: api.ServerUntrusted{
+			APIVersion: "1.1",
+		},
+	}
+
+	factory, server := lxd.NewRemoteServerFactory(ctrl)
+
+	gomock.InOrder(
+		server.EXPECT().StorageSupported().Return(false),
+		server.EXPECT().GetServer().Return(serverInfo, etag, nil),
+	)
+
+	creds := cloud.NewCredential("any", map[string]string{
+		"client-cert": "client-cert",
+		"client-key":  "client-key",
+		"server-cert": "server-cert",
+	})
+	svr, err := factory.RemoteServer(environscloudspec.CloudSpec{
+		Endpoint:   "https://10.0.0.9:8443",
+		Credential: &creds,
+	})
+	c.Assert(svr, gc.Not(gc.IsNil))
+	c.Assert(svr, gc.Equals, server)
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *serverIntegrationSuite) TestInsecureRemoteServerDoesNotCallGetServer(c *gc.C) {
```

Comment:
This is a test for the change. We know that `WithSkipGetServer(true)` because the absence of an expectation for the `Server.GetServer` call is satisfied.